src/go/rpk/pkg/cli/topic/trim.go

```diff
@@ -0,0 +1,279 @@
+
+	cmd.MarkFlagsMutuallyExclusive("to-hwm", "offset")
+	cmd.MarkFlagsMutuallyExclusive("from-file", "offset")
+	cmd.MarkFlagsMutuallyExclusive("from-file", "partitions")
+
+	return cmd
+}
+
+// parseOffsetArgs creates the kadm.Offsets based on the passed args via rpk
+// flags. It checks if the topic exists and defaults to all partitions in the
+// case that partitions were not provided.
+func parseOffsetArgs(ctx context.Context, adm *kadm.Client, all bool, topic, offset string, partitions []int) (kadm.Offsets, error) {
+	// First we check if the topic exists
+	listed, err := adm.ListTopics(ctx, topic)
+	if err != nil {
+		return nil, fmt.Errorf("unable to list topic %q metadata: %v", topic, err)
+	}
+	listed.EachError(func(d kadm.TopicDetail) {
+		err = fmt.Errorf("unable to list the topic %q: %v", topic, d.Err)
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// If no partitions are provided, we trim the record to the
+	// provided offset in all partitions.
+	if len(partitions) == 0 {
+		listed.EachPartition(func(pd kadm.PartitionDetail) {
+			partitions = append(partitions, int(pd.Partition))
+		})
+	}
+
+	o := make(kadm.Offsets)
+	for _, p := range partitions {
+		var parsedOffset int64
+		if all {
+			parsedOffset = -1 // -1 maps to the partition's high watermark.
+		} else {
+			parsedOffset, err = parseOffsetFlag(ctx, adm, topic, offset, p)
+			if err != nil {
+				return nil, fmt.Errorf("unable to parse offset %v: %v", offset, err)
+			}
+		}
+		o.Add(kadm.Offset{
+			Topic:     topic,
+			Partition: int32(p),
+			At:        parsedOffset,
+		})
+	}
+	return o, nil
+}
+
+// parseOffsetFile parse the given file whether it's a json, yaml or
+// space-delimited text. File content: topic | partition | offset.
+func parseOffsetFile(fs afero.Fs, file string) (kadm.Offsets, error) {
+	parsed, err := out.ParseFileArray[struct {
+		Topic     string `yaml:"topic" json:"topic"`
+		Partition int32  `yaml:"partition" json:"partition"`
+		Offset    int64  `yaml:"offset" json:"offset"`
+	}](fs, file)
+	if err == nil {
+		o := make(kadm.Offsets)
+		for _, element := range parsed {
+			o.Add(kadm.Offset{
+				Topic:     element.Topic,
+				Partition: element.Partition,
+				At:        element.Offset,
+			})
+		}
+		return o, nil
+	}
+	return nil, err
+}
+
+// parseOffsetFlag parse the offset string that we get from the flag, it parses
+// a simple integer or a timestamp-based offset (prefixed with @) it follows the
+// same syntax/decode as 'rpk topic consume --offset' flag.
+func parseOffsetFlag(ctx context.Context, adm *kadm.Client, topic, offset string, partition int) (int64, error) {
+	// Timestamp offset
+	if strings.HasPrefix(offset, "@") {
+		offset = offset[1:] // strip @
+		now := time.Now()
+		_, startAt, end, _, err := parseTimestampBasedOffset(offset, time.Now())
+		if err != nil {
+			return 0, err
+		} else if end {
+			return 0, errors.New("invalid offset '@end'; use '--to-hwm' if you want to trim all")
```

Comment:
I think it might be ok to allow @end -- I was thinking to block -1 only because users shouldn't need to know special numbers

Reply:
Changed to allow `--offset end` and removed `--to-hwm`